pkg/handler/handler.go

```diff
@@ -230,3 +233,49 @@ func (cs contextSource) UpdateEntityAttributes(entityID string, req ngsi.Request
 func (cs contextSource) RetrieveEntity(entityID string, req ngsi.Request) (ngsi.Entity, error) {
 	return nil, errors.New("RetrieveEntity is not supported")
 }
+
+type ApiKey struct {
+	enabled bool
+	key     string
+}
+
+func newApiKeyMiddleware() *ApiKey {
+	a := &ApiKey{
+		enabled: false,
+		key: "",
+	}
+
+	if b, err := strconv.ParseBool(os.Getenv("DIWISE_REQUIRE_API_KEY")); err == nil {
+		if b {
+			a.enabled = true
+			validKey := os.Getenv("DIWISE_API_KEY")
+
+			if len(validKey) != 0 {
+				a.key = validKey
+			} else {
+				a.enabled = false
+			}
+		}
+	} else {
+		a.enabled = false
+	}
+
+	return a
+}
+
+func (a *ApiKey) Handler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {								
+		if a.enabled && strings.ToUpper(r.Method) == "POST" {
+			
+			apiKey := r.Header.Get("x-api-key")
+			
+			if len(apiKey) == 0 || apiKey != a.key {
+				w.Header().Add("Content-Type", "application/json")
```

Comment:
NGSI-LD specar application/problem+json från RFC-7807 som content type på felsvar. Spana in https://blog.restcase.com/rest-api-error-handling-problem-details-response/ och lite exempel i ngsi-ld-golang på hur vi hanterar det.